core/src/main/java/io/questdb/griffin/WhereClauseParser.java

```diff
@@ -1158,6 +1181,64 @@ private boolean removeWithin(AliasTranslator translator, ExpressionNode node, Re
         }
     }
 
+    private void processArgument(
+            ExpressionNode inArg,
+            RecordMetadata metadata,
+            FunctionParser functionParser,
+            SqlExecutionContext executionContext,
+            int columnType,
+            LongList prefixes
+    ) throws SqlException, NumericException {
+        if(isNull(inArg)) {
+            throw SqlException.$(inArg.position, "GeoHash value expected");
+        }
+        if (isFunc(inArg)) {
+            Function f = functionParser.parseFunction(inArg, metadata, executionContext);
+            if (isGeoHashConstFunction(f)) {
+                final int fnType = f.getType();
+                final long hash = f.getGeoHashLong(null);
```

Comment:
Functions will / should not allow to call wrong type method. Some geohash resulting functions like CAST(String, Geohash) will throw assertion if you do `getGeoHashLong` on  say 7bit geohash type